jnosql-dynamodb/src/main/java/org/eclipse/jnosql/databases/dynamodb/communication/DefaultDynamoDBDocumentManager.java

```diff
@@ -0,0 +1,348 @@
+import software.amazon.awssdk.services.dynamodb.model.ExecuteStatementResponse;
+import software.amazon.awssdk.services.dynamodb.model.KeySchemaElement;
+import software.amazon.awssdk.services.dynamodb.model.KeyType;
+import software.amazon.awssdk.services.dynamodb.model.ProvisionedThroughput;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.dynamodb.model.ScalarAttributeType;
+import software.amazon.awssdk.services.dynamodb.model.ScanRequest;
+import software.amazon.awssdk.services.dynamodb.model.Select;
+import software.amazon.awssdk.services.dynamodb.model.StreamSpecification;
+import software.amazon.awssdk.services.dynamodb.model.TimeToLiveStatus;
+import software.amazon.awssdk.services.dynamodb.model.UpdateItemRequest;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static java.util.Objects.requireNonNull;
+import static org.eclipse.jnosql.databases.dynamodb.communication.DynamoDBConverter.entityAttributeName;
+import static org.eclipse.jnosql.databases.dynamodb.communication.DynamoDBConverter.toAttributeValue;
+import static org.eclipse.jnosql.databases.dynamodb.communication.DynamoDBConverter.toDocumentEntity;
+import static org.eclipse.jnosql.databases.dynamodb.communication.DynamoDBConverter.toItem;
+import static org.eclipse.jnosql.databases.dynamodb.communication.DynamoDBConverter.toItemUpdate;
+
+public class DefaultDynamoDBDocumentManager implements DynamoDBDocumentManager {
+
+    private final String database;
+
+    private final Settings settings;
+
+    private final DynamoDbClient dynamoDbClient;
+
+    private final ConcurrentHashMap<String, Supplier<String>> ttlAttributeNamesByTable = new ConcurrentHashMap<>();
+
+    private final ConcurrentHashMap<String, DescribeTableResponse> tables = new ConcurrentHashMap<>();
+
+    public DefaultDynamoDBDocumentManager(String database, DynamoDbClient dynamoDbClient, Settings settings) {
+        this.settings = settings;
+        this.database = database;
+        this.dynamoDbClient = dynamoDbClient;
+    }
+
+    private String resolveEntityNameAttributeName(String entityName) {
+        return this.settings.get(DynamoDBConfigurations.ENTITY_PARTITION_KEY, String.class).orElse(entityName);
+    }
+
+    public DynamoDbClient dynamoDbClient() {
+        return dynamoDbClient;
+    }
+
+    @Override
+    public String name() {
+        return database;
+    }
+
+    @Override
+    public DocumentEntity insert(DocumentEntity documentEntity) {
```

Comment:
we merged and the DocumentEntity became CommunicationEntity

Reply:
Fixed!